### ISSUE TITLE
HomeAssistant -> Народный монитор

### ISSUE THREAD

**Issue by diemon24** [Limych/ha-narodmon#109]
Предлагаю добавить возможность выдачи показаний собственных сенсоров из HA в Народный монитор

**Comment by deathfrozen** [comment]
Да, тоже зашел сюда - посмотреть есть ли такая возможность, чтобы свой велосипед не городить.

**Comment by HEMO74** [comment]
+1

**Comment by Myumdom** [comment]
как этуой шляпой передать с ха показания на карту нар мон? что так через одно место все сденлано? попроще никак нельзя?

**Comment by Limych** [comment]
> как этуой шляпой передать с ха показания на карту нар мон? что так через одно место все сденлано? попроще никак нельзя?

Можно, конечно.
Давайте Вы напишете патч к этому компоненту, который будет реализовывать нужный функционал максимально просто. Многие люди Вам скажут спасибо, я уверен.

**Comment by DimChik2903** [comment]
Очень было бы полезно. Раньше пользовался решением с appdaemon, но оно перестало работать в ноябре 2023(

**Comment by SNMetamorph** [comment]
@DimChik2903 вот есть некое решение для этого: https://github.com/Lefey/ad_narodmon_sender
Но пока ничего сказать не могу, не тестировал.

**Comment by DimChik2903** [comment]
> [@DimChik2903](https://github.com/DimChik2903) вот есть некое решение для этого: https://github.com/Lefey/ad_narodmon_sender Но пока ничего сказать не могу, не тестировал.

Ну я же в сообщении выше написал, что решение с appdaemon перестало у меня работать) Сейчас отправляю показания с ESPHome, там работает как часы. Но немного костыльно, HA собирает показания с разных датчиков в сенсор, одна из ESP берёт этот сенсор из HA и шлёт его в народмон.

**Comment by diemon24** [comment]
> > [@DimChik2903](https://github.com/DimChik2903) вот есть некое решение для этого: https://github.com/Lefey/ad_narodmon_sender Но пока ничего сказать не могу, не тестировал.
> 
> Ну я же в сообщении выше написал, что решение с appdaemon перестало у меня работать) Сейчас отправляю показания с ESPHome, там работает как часы. Но немного костыльно, HA собирает показания с разных датчиков в сенсор, одна из ESP берёт этот сенсор из HA и шлёт его в народмон.

А чего умеет еспх такого, что не умеет ХА?

**Comment by DimChik2903** [comment]
> А чего умеет еспх такого, что не умеет ХА?

Ничего такого, там простая отправка json. Это я просто не умею нормально из ХА это отправлять, а для ESPHome нашёл готовый код и скопировал себе.

**Comment by diemon24** [comment]
> > А чего умеет еспх такого, что не умеет ХА?
> 
> Ничего такого, там простая отправка json. Это я просто не умею нормально из ХА это отправлять, а для ESPHome нашёл готовый код и скопировал себе.

Можете сюда плс его опубликовать?

**Comment by DimChik2903** [comment]
> Можете сюда плс его опубликовать?

```
# Отправка датчиков на народный мониторинг
http_request:
  id: narodmon_request
  timeout: 15s
  verify_ssl: false

interval:
  - interval: 6min
    then:
      - lambda: |-
          DynamicJsonDocument doc(1024);
          bool has_data = false;

          JsonObject device = doc.createNestedArray("devices").createNestedObject();
          device["mac"] = "МАКАДРЕСУСТРОЙСТВА";
          device["name"] = "НАЗВАНИЕ";
          device["owner"] = "ЛОГИН";
          device["lat"] = 55.5555;
          device["lon"] = 33.3333;
          JsonArray sensors = device.createNestedArray("sensors");

          if (!isnan(id(outdoor_temp_ha).state)) {
            JsonObject temp = sensors.createNestedObject();
            temp["id"] = "TEMP";
            temp["value"] = id(outdoor_temp_ha).state;
            temp["unit"] = "C";
            temp["name"] = "Улица";
            has_data = true;
          }

          if (!isnan(id(outdoor_pressure_ha).state)) {
            JsonObject pressure = sensors.createNestedObject();
            pressure["id"] = "PRESSURE";
            pressure["value"] = id(outdoor_pressure_ha).state;
            pressure["unit"] = "hPa";
            pressure["name"] = "Давление";
            has_data = true;
          }

          if (has_data) {
            std::string payload;
            serializeJson(doc, payload);
            id(narodmon_request).post("https://narodmon.ru/json", payload);
          } else {
            ESP_LOGW("narodmon", "No valid sensor data, skipping upload.");
          }
```

**Comment by diemon24** [comment]
> > Можете сюда плс его опубликовать?
> 
> ```
> # Отправка датчиков на народный мониторинг
> http_request:
>   id: narodmon_request
>   timeout: 15s
>   verify_ssl: false
> 
> interval:
>   - interval: 6min
>     then:
>       - lambda: |-
>           DynamicJsonDocument doc(1024);
>           bool has_data = false;
> 
>           JsonObject device = doc.createNestedArray("devices").createNestedObject();
>           device["mac"] = "МАКАДРЕСУСТРОЙСТВА";
>           device["name"] = "НАЗВАНИЕ";
>           device["owner"] = "ЛОГИН";
>           device["lat"] = 55.5555;
>           device["lon"] = 33.3333;
>           JsonArray sensors = device.createNestedArray("sensors");
> 
>           if (!isnan(id(outdoor_temp_ha).state)) {
>             JsonObject temp = sensors.createNestedObject();
>             temp["id"] = "TEMP";
>             temp["value"] = id(outdoor_temp_ha).state;
>             temp["unit"] = "C";
>             temp["name"] = "Улица";
>             has_data = true;
>           }
> 
>           if (!isnan(id(outdoor_pressure_ha).state)) {
>             JsonObject pressure = sensors.createNestedObject();
>             pressure["id"] = "PRESSURE";
>             pressure["value"] = id(outdoor_pressure_ha).state;
>             pressure["unit"] = "hPa";
>             pressure["name"] = "Давление";
>             has_data = true;
>           }
> 
>           if (has_data) {
>             std::string payload;
>             serializeJson(doc, payload);
>             id(narodmon_request).post("https://narodmon.ru/json", payload);
>           } else {
>             ESP_LOGW("narodmon", "No valid sensor data, skipping upload.");
>           }
> ```

Спасибо! Почему бы такую функционал не добавить прям в интеграции народмона?... предлагаю разработчикам рассмотреть работающий пример

**Comment by MERKAT0R** [comment]
Как тут выше заметили, в плане отправки на сервис дополнение _сейчас_ весьма бестолковое(да и в принципе в целом, если честно). Кодом ниже это можно реализовать для своих датчиков прямо в HASSе .

**configuration.yaml**

```
конфига больше тут нет - никому не советуют пользоваться данным сервисом, владелец неадекват
```


**automations.yaml**

```
# Upload sensors to NarodMon Automation
- id: narodmon_send_simple
  alias: "NarodMon — отправка каждые 6 мин"
  trigger:
    - platform: time_pattern
      minutes: "/6"
  condition: []
  action:
    - service: rest_command.narodmon_send
      data: {}
  mode: single
```

**Comment by adonixis** [comment]
> конфига больше тут нет - никому не советуют пользоваться данным сервисом, владелец неадекват

о каком именно сервисе идет речь?
